digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/RoadAddressService.scala

```diff
@@ -526,7 +529,7 @@ class RoadAddressService(roadLinkService: RoadLinkService, eventbus: DigiroadEve
 
     val addresses =
       withDynTransaction {
-        RoadAddressDAO.fetchByLinkId(roadLinksWithComplimentary.map(_.linkId).toSet, false, false).groupBy(_.linkId)
+        RoadAddressDAO.fetchByLinkId(roadLinksWithComplimentary.map(_.linkId).toSet, false, false).filterNot(ra => ra.endDate.isDefined).groupBy(_.linkId)
```

Comment:
It's better to do the filtering in a query and not to fetch any database rows for filtering in the application code.

Reply:
I'll remove this validation, since we already have a boolean in the method fetchByLinkId that indicates if we want history of the road address and validates if the end date is null